File: Android-Core/01-AndroidCore/a-Activity/03_ActivityBasic/app/src/main/java/top/iqqcode/activitybasic/SecondActivity.java
package top.iqqcode.activitybasic;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

public class SecondActivity extends AppCompatActivity {

    public static final String ACTION = "com.iqqcode.activitybasic.intent.action.SecondActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
    }
}

File: Kotlin/a-Basic/ActivityDemo/app/src/main/java/top/iqqcode/activitydemo/SecondActivity.kt
package top.iqqcode.activitydemo

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import top.iqqcode.activitydemo.databinding.ActivitySecondBinding

class SecondActivity : AppCompatActivity() {

    private lateinit var binding: ActivitySecondBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySecondBinding.inflate(layoutInflater)
        setContentView(binding.root)

        // extras?  为null则不执行getString("name")
        val name = intent.getStringExtra("name")
        binding.textView.text = name

        val user = intent.getParcelableExtra<User>("user")
        binding.textViewContext.text = user?.username + " " + user?.password.toString()

        binding.button.setOnClickListener {
            Intent(this, ThirdActivity::class.java).apply {
                putExtras(intent.extras!!)
                putExtra("date", "2021-04-19")
                startActivity(this)
            }
        }
    }
}